Redirect users after login and add docs link to dashboard

satyam1024 opened the pull request:
### Description: 
Redirects authenticated users to the dashboard instead of keeping them on the login page. 
Adds a documentation link on the dashboard for user guidance. 
Updated dashboard styling . 

### Why These Changes? 
✅ Better UX – Users are redirected automatically after login.
✅ Faster Onboarding – The docs link helps users learn W-stack easily.
✅ Improved UI – Dashboard now has better styling for a polished look.

### Changes Made: 
Implemented login redirection. 
Added a docs link to the dashboard.


![Screenshot (3198)](https://github.com/user-attachments/assets/e6e43e1e-d68f-449e-896c-8358ee00efc1)

satyam1024 committed:
fix(dashboard and login page): redirect users after login and add docs link to dashboard

Implemented automatic redirection to the dashboard after successful login.Added a link to the Wstack
documentation on the dashboard for user guidance.Improved user experience by ensuring authenticated
users don’t stay on the login page.

satyam1024 committed:
revert(.env.example and package-lock.json): recreating .env.example file and package-lock file

since at the time of this commit package-lock.json file is empty so to not create any conflict i
deleted content of my package-lock file.

nameerakhter commented:
can you please add some screenshots 

Always attach screenshots whenever you make a issue or a PR

satyam1024 commented:
sorry about that. I will keep that in mind going forward.

nameerakhter commented on src/app/(auth)/login/page.tsx:
```diff
@@ -1,26 +1,28 @@
 
+  const router = useRouter()
+
+  useEffect(() => {
```
Sir docs links kari h toh padh bhi lete  🫠

Please try to remove useEffect,  just dont use it unless and until absoultely necessary

satyam1024 replied:
Is it alright to use redirect() here.

nameerakhter commented on src/app/(protected)/dashboard/page.tsx:
```diff
@@ -18,20 +18,37 @@ export default function Dashboard() {
-  )
+
+
+    return (
```
Not a super important PR tbh but fine fix the small changes I'll merge it for now

Ankur5522 replied:
in this also don't hard code the callback url. make a env variable for it as after we deploy it we would need to change it.

satyam1024 replied:
Good morning.
I have created an env variable for the callback, but I am unsure about the best way to redirect the user. Could you give me a hint?

satyam1024 committed:
refactor(.env.example): made an env variable for Documentation Url

satyam1024 committed:
chore(package-lock.json): removed all content

satyam1024 committed:
chore(dashboard page): remove unwanted console.log()

satyam1024 committed:
fix(.env.example ,login page): created a env variable for callback url

satyam1024 committed:
fix(src): added middleware.ts to handle redirects

Added middleware.ts to auto redirect user from login to dashboard after authentication and prevent
users to access protected pages without login.

iamDyeus commented:
Hey there, the repo has undergone some changes. Please fix the branch conflicts and then request a review again.
